pysages/colvars/contacts.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: MIT
+# See LICENSE.md and CONTRIBUTORS.md at https://github.com/SSAGESLabs/PySAGES
+
+"""
+Collective variable related to native contacts.
+
+Q describes the fraction of the native contacts
+for folding of proteins or nucleic acids
+"""
+from jax import numpy as np
+
+from pysages.colvars.core import CollectiveVariable, multicomponent
+
+
+def remap_indices(contacts, sorted_unique_indices):
+    # create the mapping from old indices to new indices
+    old_indices = np.array(sorted_unique_indices)
+    new_indices = np.arange(len(sorted_unique_indices))
+    index_mapping = np.zeros(old_indices.max() + 1, dtype=new_indices.dtype)
+    index_mapping = index_mapping.at[old_indices].set(new_indices)
+
+    contacts = np.array(contacts)
+    remapped_contacts = index_mapping[contacts]
+
+    return remapped_contacts
```

Comment:
I need to check, but this looks like something that could be accomplished, by using `argsort`